app/models/mixins/cloud_tenancy_mixin.rb

```diff
@@ -0,0 +1,19 @@
+module CloudTenancyMixin
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    include TenancyCommonMixin
+
+    def tenant_id_clause_format(tenant_ids)
+      ["(tenants.id IN (?) AND ext_management_systems.tenant_mapping_enabled IS TRUE) OR ext_management_systems.tenant_mapping_enabled IS FALSE", tenant_ids]
+    end
+
+    def tenant_joins_clause(scope)
+      scope.includes(:cloud_tenant => "source_tenant").joins(:ext_management_system)
+    end
+  end
+
+  def tenant
```

Comment:
we are using it somewhere? where ?
if so shouldn't it be `cloud_tenant.try(:source_tenant)`